src/tox/package.py

```diff
@@ -83,3 +99,113 @@ def make_sdist(report, config, session):
                 "     python setup.py sdist"
             )
             raise SystemExit(1)
+
+
+def build_isolated(config, report, session):
+    build_info = get_build_info(config.setupdir, report)
+    package_venv = session.getvenv(".package")
+    package_venv.envconfig.deps_matches_subset = True
+
+    package_venv.envconfig.deps = [DepConfig(r, None) for r in build_info.requires]
+    toml_require = {pkg_resources.Requirement(r).key for r in build_info.requires}
+    if not session.setupenv(package_venv):
+        raise SystemExit(1)
+
+    live_config = package_venv._getliveconfig()
+    previous_config = CreationConfig.readconfig(package_venv.path_config)
+    if not previous_config or not previous_config.matches(live_config, True):
+        session.finishvenv(package_venv)
+
+    build_requires = get_build_requires(build_info, package_venv, session)
+    for requirement in build_requires:
+        pkg_requirement = pkg_resources.Requirement(requirement)
+        if pkg_requirement.key not in toml_require:
+            package_venv.envconfig.deps.append(DepConfig(requirement, None))
+
+    if not session.setupenv(package_venv):
```

Comment:
here we are recreating the env maybe instead just install new dependencies and finalize again the env